src/zero_copy/mod.rs

```diff
@@ -74,6 +75,80 @@ use self::{
 /// The result of calling [`Parser::go`]
 pub type PResult<M, O, E> = Result<<M as Mode>::Output<O>, Located<E>>;
 
+/// The result of running a [`Parser`]. [`ParseResult::Warn`] and [`ParseResult::Err`] vectors
+/// will always have at least one element.
+#[derive(Debug, Clone, PartialEq)]
+pub enum ParseResult<T, E> {
+    /// Parser succeeded with no issues
+    Ok(T),
+    /// Parser succeeded with errors (hit a recovery path, for example)
+    Warn(T, Vec<E>),
+    /// Parser failed with errors
+    Err(Vec<E>),
+}
```

Comment:
Hmm, I'm unconvinced that an enum is the way to go. I think I'd prefer to see something like

```rs
struct ParseResult<T, E> {
    output: Option<T>,
    errors: Vec<E>,
}
```

That way, we can hide the implementation and add features like actual warnings (i.e: not errors) in the future without it being a breaking change.